test: failure in interface_usdt_validation.py

https://cirrus-ci.com/task/5553601580892160?logs=ci#L4244:
```bash
 test  2022-08-01T12:14:24.104000Z TestFramework (INFO): handle_blockconnected(): ConnectedBlock(hash=2f97f65fda63aba081bf2867f7f99657c39436306c4aed3c899264b8bde7c5ab height=202, transactions=1, inputs=1, sigops=0, duration=517) 
 test  2022-08-01T12:14:24.104000Z TestFramework (INFO): handle_blockconnected(): ConnectedBlock(hash=54d4bd414e5de85cee4dc5f1e7bcf4e358cf6e7ce556647debe2e9d007459ec7 height=201, transactions=1, inputs=1, sigops=0, duration=338) 
 test  2022-08-01T12:14:24.148000Z TestFramework (INFO): check that we traced 2 blocks 
 test  2022-08-01T12:14:24.149000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 133, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/interface_usdt_validation.py", line 131, in run_test
                                       assert_equal(BLOCKS_EXPECTED, blocks_checked)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 52, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(2 == 0)
```

--- a/test/functional/interface_usdt_validation.py
+++ b/test/functional/interface_usdt_validation.py
@@ -91,7 +91,7 @@ class ValidationTracepointTest(BitcoinTestFramework):
         # that the handle_* functions succeeded.
         BLOCKS_EXPECTED = 2
         blocks_checked = 0
-        expected_blocks = list()
+        expected_blocks = dict()
 
         self.log.info("hook into the validation:block_connected tracepoint")
         ctx = USDT(pid=self.nodes[0].process.pid)
@@ -104,15 +104,16 @@ class ValidationTracepointTest(BitcoinTestFramework):
             nonlocal expected_blocks, blocks_checked
             event = ctypes.cast(data, ctypes.POINTER(Block)).contents
             self.log.info(f"handle_blockconnected(): {event}")
-            block = expected_blocks.pop(0)
-            assert_equal(block["hash"], bytes(event.hash[::-1]).hex())
+            block_hash = bytes(event.hash[::-1]).hex()
+            block = expected_blocks[block_hash]
+            assert_equal(block["hash"], block_hash)
             assert_equal(block["height"], event.height)
             assert_equal(len(block["tx"]), event.transactions)
             assert_equal(len([tx["vin"] for tx in block["tx"]]), event.inputs)
             assert_equal(0, event.sigops)  # no sigops in coinbase tx
             # only plausibility checks
             assert(event.duration > 0)
-
+            del expected_blocks[block_hash]
             blocks_checked += 1
 
         bpf["block_connected"].open_perf_buffer(
@@ -122,7 +123,7 @@ class ValidationTracepointTest(BitcoinTestFramework):
         block_hashes = self.generatetoaddress(
             self.nodes[0], BLOCKS_EXPECTED, ADDRESS_BCRT1_UNSPENDABLE)
         for block_hash in block_hashes:
-            expected_blocks.append(self.nodes[0].getblock(block_hash, 2))
+            expected_blocks[block_hash] = self.nodes[0].getblock(block_hash, 2)
 
         bpf.perf_buffer_poll(timeout=200)
         bpf.cleanup()

